### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/type/StandardBasicTypes.java [hunk 1]
```diff
@@ -539,9 +539,8 @@ private StandardBasicTypes() {
 	/**
 	 * The standard Hibernate type for mapping {@link Byte Byte[]} to JDBC {@link org.hibernate.type.SqlTypes#VARBINARY VARBINARY}.
 	 */
-	public static final BasicTypeReference<Byte[]> WRAPPER_BINARY = new BasicTypeReference<>(
-			//TODO find a decent name before documenting
-			"wrapper-binary",
+	public static final BasicTypeReference<Byte[]> BINARY_WRAPPER = new BasicTypeReference<>(
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/type/StandardBasicTypes.java:542]: I'm not following these changes...

Users would use these names in their mappings.  But their old mappings would no longer match.  Surely I am missing something right?

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/type/StandardBasicTypes.java:542]: Oh yes, you are right. That one was renamed by "mistake", thanks for pointing it out.

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/type/StandardBasicTypes.java:542]: Its not just one though.  Just look throughout the diff for this file

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/type/StandardBasicTypes.java:542]: Bottom line, while you may not like the names - they are exposed to users and cannot be changed.  Add a new "synonym" if you wish, but do not change them

**Pull request merged**